.github/workflows/prepare-release.yml

```diff
@@ -0,0 +1,57 @@
+---
+# test with
+# act --job bump-crate-version --eventpath ./.github/example/prepare-release.event
+name: prepare release
+on:
+  workflow_dispatch:
+    inputs:
+      versionBump:
+        description: "Version bump for"
+        required: true
+        type: choice
+        options:
+          - major
+          - minor
+          - patch
+
+jobs:
+  prepare-release:
+    name: Prepare release
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v5
+        with:
+          fetch-tags: true
+
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@stable
+
+      - name: Bump versions
+        env:
+          FRAGMENT: ${{ github.event.inputs.versionBump }}
+        run: ./.github/scripts/bump-versions.sh
+
+      - name: Update Cargo.lock
+        run: cargo update --workspace
+
+      - name: Get binary version
+        id: get-version
+        run: |
+          VERSION=$(cat ./Cargo.toml | awk "/version/{print; exit}" | cut -d\" -f 2)
+          echo VERSION=$VERSION >> ${GITHUB_OUTPUT}
+
+      - name: Update Changelog
+        uses: thomaseizinger/keep-a-changelog-new-release@3.1.0
+        with:
+          tag: v${{ steps.get-version.outputs.VERSION }}
+          version: ${{ steps.get-version.outputs.VERSION }}
+
+      - name: Commit version prepare
+        uses: stefanzweifel/git-auto-commit-action@v6
+        if: ${{ !env.ACT }}
```

Comment:
Could we check more generally for development environments besides ACT?

Reply:
And how would that look like? Or what different development environment do you have in mind?

Reply:
Yeah don't know how we can make that work with GitHub actions? Not an expert.

But if we would have something like:

```yaml
if: ${{ env.PRODUCTION }}
```

then that could be more robust.

Can we set that environment variable somehow?

Reply:
Ah good point restricting it from the other side. I will take a look if there is something github provides.

Even tho I think this will break the intention of not running this step locally when using act, but I will report about that when I looked into it.

Reply:
So I tried to find something similar that is supported by github out of the box, but I couldn't find anything. So only disabling it when using act seems good enough from my perspective.